### PR TITLE
Feature/count vectorizer

### PR DESCRIPTION
**Pull request by steveoni**
This pull request implement countVectorizer.

```js
const vectorizer = new CountVectorizer(["The quick brown fox jumped over the lazy dog."]);
const toVector = vectorizer.transform(["the brown brown","the the brown"]);

console.log(toVector)
// output
// [
//  [2, 0, 0, 0,0, 0, 0, 1],
//  [1, 0, 0, 0,0, 0, 0, 2]
//  ]

console.log(vectorizer.wordOrder)
//output
// {
//    brown: 0,
//    dog: 1,
//    fox: 2,
//    jumped: 3,
//    lazy: 4,
//    over: 5,
//    quick: 6,
//    the: 7
//  }

```

### CODE REVIEW

File: src/text/countvectorizer.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  /**
+   * Convert text to vector base on their number of count
+   * and the order to which they occur alphabetically.
+   * @param textArray
+   */
+  constructor(textArray: string[]) {
+
+    const tokenArray: string[] = [];
+    textArray.forEach((value) => {
+      value.split(" ").forEach((text) => {
```
Comment (yorkie) [inline src/text/countvectorizer.ts:18]:
```suggestion
      value.split(' ').forEach((text) => {
```

File: src/text/countvectorizer.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        if (text != '') {
```
Comment (yorkie) [inline src/text/countvectorizer.ts:19]:
```suggestion
        if (text !== '') {
```

File: src/text/countvectorizer.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    });
+
+    const uniqueWord = Array.from(new Set(tokenArray));
+    const sortedUniqueWord: string[] = this.sort(uniqueWord);
+
+    // store array element as object element
+    // to enable quick search of element
+    // when transforming input arrays.
+    sortedUniqueWord.forEach((text, index) => {
+      if (text != '') {
```
Comment (yorkie) [inline src/text/countvectorizer.ts:32]: Ditto

File: src/text/countvectorizer.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        this.wordOrder[text] = index;
+      }
+    });
+    this.uniqueLength = sortedUniqueWord.length;
+
+  }
+
+  private sort(textArray: string[]): string[] {
+    const collator = new Intl.Collator(undefined, { numeric:true, sensitivity: 'base' });
```
Comment (yorkie) [inline src/text/countvectorizer.ts:41]: Shall we have a way to downgrade if Intl is unavailable?

File: src/text/countvectorizer.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+   * Transform input text to vector.
+   * @param textArray
+   * @returns number[][] array of number (vectors)
+   */
+  public transform(textArray: string[]): number[][] {
+
+    const counterVectorizer: number[][] = textArray.map((value) => {
+      const innerArray: number[] = Array.from(new Float64Array(this.uniqueLength));
+      const cleanTextArray: string[] = [];
+      value.split(" ").forEach((text) => {
```
Comment (yorkie) [inline src/text/countvectorizer.ts:55]: Ditto, we need to unify the format to use quotes :)

File: test/node/text/countvectorizer.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    const vocab = {
+                    brown: 0,
+                    dog: 1,
+                    fox: 2,
+                    jumped: 3,
+                    lazy: 4,
+                    over: 5,
+                    quick: 6,
+                    the: 7
+                  }
```
Comment (yorkie) [inline test/node/text/countvectorizer.ts:17]:
```suggestion
                  };
```

File: test/node/text/countvectorizer.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    const expectedVector = [
+                            [2, 0, 0, 0,0, 0, 0, 1],
+                            [1, 0, 0, 0,0, 0, 0, 2]
+                           ]
```
Comment (yorkie) [inline test/node/text/countvectorizer.ts:21]:
```suggestion
                           ];
```

**Review by yorkie — CHANGES_REQUESTED** [review]

File: src/text/countvectorizer.ts (hunk 4 above)
Comment (steveoni) [reply src/text/countvectorizer.ts:41]: ok

File: test/node/text/countvectorizer.ts (hunk 7 above)
Comment (steveoni) [reply test/node/text/countvectorizer.ts:21]: ok

File: test/node/text/countvectorizer.ts (hunk 6 above)
Comment (steveoni) [reply test/node/text/countvectorizer.ts]: ok

File: src/text/countvectorizer.ts (hunk 5 above)
Comment (steveoni) [reply src/text/countvectorizer.ts]: ok

**Comment by steveoni** [comment]
why is it showing`Merging is blocked`? request changes by @yorkie are now addressed in the new push

**Review by yorkie — APPROVED** [review]

**Comment by yorkie** [comment]
@steveoni sorry, I have approved this :)

**Comment by steveoni** [comment]
> @steveoni sorry, I have approved this :)

ok. I was confused though because normally after making some changes based on feedback, it does not show merging is blocked.

@yorkie

**Comment by yorkie** [comment]
Yea, PR will be blocked util the change request is removed, I will update ASAP next time :)

**Pull request merged**